salt/state.py

```diff
@@ -2308,10 +2309,17 @@ def _get_envs(self):
         '''
         Pull the file server environments out of the master options
         '''
-        envs = set(['base'])
+        envs = ['base']
         if 'file_roots' in self.opts:
-            envs.update(list(self.opts['file_roots']))
-        return envs.union(set(self.client.envs()))
+            envs.extend(list(self.opts['file_roots']))
+        client_envs = self.client.envs()
+        if self.opts.get('env_order', []):
```

Comment:
We should put a check here to not include envs that do not exist in the return from the self.client.envs() call